### PR TITLE
Resume media

### PR DESCRIPTION
**Pull request by laiyichin**
#### WHAT
Resume last played media.

#### WHY
If a user opted in, they can have the media resumed.

#### HOW
- Write to SharedPreferences in MediaPlayerScreenViewModel at the 1 second tick
- Read the SharedPreferences if intent was not set (from tiles) and the dev option loadItemsAtStartup is set to true.
- Further improvement needed and documented at https://github.com/google/horologist/issues/684

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

resume_media

### CODE REVIEW

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt [hunk 1]
```diff
@@ -52,12 +54,22 @@ import com.google.android.horologist.mediasample.ui.playlists.UampPlaylistsScree
...
 import com.google.android.horologist.mediasample.ui.settings.DeveloperOptionsScreen
 import com.google.android.horologist.mediasample.ui.settings.UampSettingsScreen
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
 
 @Composable
 fun UampWearApp(
+    context: Context,
     navController: NavHostController,
+    loadItemsAtStartupFlow: Flow<Boolean>,
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt:64]: https://medium.com/androiddevelopers/consuming-flows-safely-in-jetpack-compose-cde014d0d5a3

Suggestion: this composable has a MediaPlayerAppViewModel which has SettingsRepository injected in it. Expose `loadItemsAtStartupFlow` in MediaPlayerAppViewModel, then UampWearApp can access it

File: media-data/src/main/java/com/google/android/horologist/media/data/repository/PlayerRepositoryImpl.kt [hunk 2]
```diff
@@ -408,4 +410,16 @@ public class PlayerRepositoryImpl(
...
+
+    override fun setCurrentMediaListId(mediaListId: String) {
+        _currentMediaListId.value = mediaListId
+    }
+
+    public fun getCurrentMediaListId(): String? {
+        return _currentMediaListId.value
+    }
+
+    public fun getCurrentMediaPosition(): Long? {
```
[the remark this answers is not in the document]
Comment (luizgrp) [reply media-data/src/main/java/com/google/android/horologist/media/data/repository/PlayerRepositoryImpl.kt:422]: actually this function is also redundant, given there is a [mediaPosition](https://github.com/google/horologist/blob/c3755f7d3eca371037b63f5fa498263e9c0225fc/media/src/main/java/com/google/android/horologist/media/repository/PlayerRepository.kt#L53)

[the remark this answers is not in the document]
Comment (laiyichin) [reply media-data/src/main/java/com/google/android/horologist/media/data/repository/PlayerRepositoryImpl.kt:422]: Removed.

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt (hunk 1 above)
Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt:64]: Done.

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/player/MediaPlayerScreenViewModel.kt [hunk 3]
```diff
@@ -33,15 +36,27 @@ import kotlin.time.Duration.Companion.seconds
 
 @HiltViewModel
 class MediaPlayerScreenViewModel @Inject constructor(
+    @ApplicationContext context: Context,
```
[the remark this answers is not in the document]
Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/player/MediaPlayerScreenViewModel.kt:39]: Will keep these in mind in the future. Removing the context for now in favour of datastore (over shared preferences).

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt [hunk 4]
```diff
@@ -198,8 +198,8 @@ fun UampWearApp(
     }
 
     LaunchedEffect(Unit) {
-        val collectionId = intent.getAndRemoveKey(MediaActivity.CollectionKey)
-        val mediaId = intent.getAndRemoveKey(MediaActivity.MediaIdKey)
+        var collectionId = intent.getAndRemoveKey(MediaActivity.CollectionKey)
+        var mediaId = intent.getAndRemoveKey(MediaActivity.MediaIdKey)
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt:202]: nit: these changes are probably not needed anymore?

**Review by luizgrp — APPROVED** [review]
LGTM!

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt (hunk 4 above)
Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/UampWearApp.kt:202]: Reverted. Thank you for catching these!

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/MediaPlayerAppViewModel.kt [hunk 5]
```diff
@@ -87,7 +80,16 @@ class MediaPlayerAppViewModel @Inject constructor(
 
         val currentMediaItem = playerRepository.currentMedia.value
 
-        if (currentMediaItem == null) {
+        // If it's currently not playing and user opted in to load items at startup,
+        // then we start playing using the last played media item.
+        if (currentMediaItem == null &&
+            settingsRepository.settingsFlow.map { it.loadItemsAtStartup }.first()) {
```
Comment (yschimke) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/MediaPlayerAppViewModel.kt]: It's worth pulling this out into a variable.

```
val settings = settingsRepository.settingsFlow.first()
```

Then reusing as `settings.loadItemsAtStartup`, `settings.currentMediaItemId` and `settings.currentMediaListId`

Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/app/MediaPlayerAppViewModel.kt]: Done. This is much more succinct. Thank you!

**Pull request merged**